File: PR1/PR1_Ej1/src/FactoriaCarretera.java
public class FactoriaCarretera implements FactoriaCarreraYBicicleta{
    @Override
    public Carrera crearCarrera() {
        return new CarreraCarretera();
    }

    @Override
    public Bicicleta crearBicicleta(int id) {
        return new BicicletaCarretera(id);
    }
}

File: PR1/PR1_Ej2/FactoriaCarretera.py
from CarreraCarretera import CarreraCarretera
from BicicletaCarretera import BicicletaCarretera
from FactoriaCarreraYBicicleta import FactoriaCarreraYBicicleta

class FactoriaCarretera(FactoriaCarreraYBicicleta):
    def crearCarrera(self):
        return CarreraCarretera()

    def crearBicicleta(self):
        return BicicletaCarretera()
